frontend/ui-pablo/app/components/Organisms/pool/CreatePool/steps/SetLiquidityStep.tsx

```diff
@@ -98,50 +98,26 @@ const SetLiquidityStep: React.FC<BoxProps> = ({ ...boxProps }) => {
     return new BigNumber(createPool.liquidity.quoteAmount);
   }, [createPool.liquidity.quoteAmount]);
 
-  const balance1 = useMemo(() => {
-    if (createPool.baseAsset !== "none") {
-      return new BigNumber(balances[createPool.baseAsset].picasso);
-    } else {
-      return new BigNumber(0);
-    }
-  }, [balances, createPool.baseAsset]);
-
-  const balance2 = useMemo(() => {
-    if (createPool.quoteAsset !== "none") {
-      return new BigNumber(balances[createPool.quoteAsset].picasso);
-    } else {
-      return new BigNumber(0);
-    }
-  }, [balances, createPool.quoteAsset]);
+  const balance1 = useAssetBalance(DEFAULT_NETWORK_ID, createPool.baseAsset);
+  const balance2 = useAssetBalance(DEFAULT_NETWORK_ID, createPool.quoteAsset);
 
   const [valid1, setValid1] = useState<boolean>(false);
   const [valid2, setValid2] = useState<boolean>(false);
 
-  const tokenToUSD1 = useMemo(() => {
-    if (createPool.baseAsset !== "none") {
-      const onChainId = getAssetOnChainId(
-        DEFAULT_NETWORK_ID,
-        createPool.baseAsset
-      );
-      if (onChainId && apollo[onChainId.toString()]) {
-        return new BigNumber(apollo[onChainId.toString()]);
-      }
-    }
-    return new BigNumber(0);
-  }, [createPool.baseAsset, apollo]);
+  const tokenToUSD1 = useUSDPriceByAssetId(createPool.baseAsset);
+  const tokenToUSD2 = useUSDPriceByAssetId(createPool.quoteAsset);
+
+  const _baseAsset = useMemo(() => {
+    return supportedAssets.find(i => {
+      return i.network[DEFAULT_NETWORK_ID] === createPool.baseAsset
+    })
+  }, [createPool.baseAsset, supportedAssets])
 
-  const tokenToUSD2 = useMemo(() => {
-    if (createPool.quoteAsset !== "none") {
-      const onChainId = getAssetOnChainId(
-        DEFAULT_NETWORK_ID,
-        createPool.quoteAsset
-      );
-      if (onChainId && apollo[onChainId.toString()]) {
-        return new BigNumber(apollo[onChainId.toString()]);
-      }
-    }
-    return new BigNumber(0);
-  }, [createPool.quoteAsset, apollo]);
+  const _quoteAsset = useMemo(() => {
+    return supportedAssets.find(i => {
+      return i.network[DEFAULT_NETWORK_ID] === createPool.quoteAsset
+    })
+  }, [createPool.quoteAsset, supportedAssets])
```

Comment:
This seems like something that is being used multiple times, and since all dependencies come from the store, could you expose quoteAsset from the useStore instead of writing this code every time when quoteAsset is needed?